src/jit/compiler.cpp

```diff
@@ -4191,6 +4191,16 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
 
     fgImport();
 
+    if (IsTargetAbi(CORINFO_CORERT_ABI) && hasFatPointer())
+    {
+        if (fgCheapPredsValid)
+        {
+            fgRemovePreds();
+        }
+        fgTransformFatCalli();
+        setHasFatPointer(false);
+    }
+
     assert(!fgComputePredsDone);
```

Comment:
> Also, could you put the setHasFatPointer(false) at the end of fgTransformFatCalli()? In this function, I'd prefer that as much as possible, it's just a linear listing of phases.

clearHasFatPointer in compiler.cpp after 
`if (hasFatPointer())`
looks more obviously for me, you don't need to open fgTransformFatCalli to see that this method flag is not set after this check. But I am not sure and will try your variant,